src/triage/component/catwalk/storage.py

```diff
@@ -29,17 +39,24 @@ def write(self, obj):
 
 
 class S3Store(Store):
+
     def exists(self):
-        return key_exists(self.path)
+        s3 = s3fs.S3FileSystem()
```

Comment:
It could. I think tweaks to the S3 semantics like these are probably low priority as I'm going to make more drastic changes soon when I apply my in progress changes to this branch.